### ISSUE TITLE
Server Side Rendering

### ISSUE THREAD

**Issue by jvsinghk** [jvsinghk/spacex#7]
You can add SSR to this or can take my commits in the `feature/server-side-rendering` branch and fix it.

The initial launch programs landing page ideally should be server-side-rendered.
A boilerplate to implement the Server-side rendering can be used.

**Comment by devil-cyber** [comment]
I would like to work on this can you assign this issue to me

**Comment by jvsinghk** [comment]
Sure @devil-cyber. Let me know about your approach.

**Comment by devil-cyber** [comment]
I think we can achive this bu using 
Express with babel and  webpack.
For our server code to work, we’ll need to bundle and transpile it, using webpack and Babel. To accomplish this. I am newbie to the world of react .

**Comment by jvsinghk** [comment]
Great, go ahead try out. I think it will work.

**Comment by devil-cyber** [comment]
Ok

**Comment by jvsinghk** [comment]
@devil-cyber are you working on this. If yes, you will have to fork this repo to make a remote copy of this on your profile. Then you can clone and commit to a new branch to open a pull request.

**Comment by devil-cyber** [comment]
Yes i will start tomorrow, this time i am working on some other project

**Comment by maelannb** [comment]
Hey, is this issue still open ?

**Comment by jvsinghk** [comment]
@setheal Hey, yes this issue is still open

**Comment by jvsinghk** [comment]
@devil-cyber are you planning to contribute to this today? Unassigned you because you haven't forked yet. This issue is open for contribution. @setheal would you like to work on this?

**Comment by devil-cyber** [comment]
Go ahed , sorry for late but I am  admitted in hospital from last night

**Comment by jvsinghk** [comment]
@devil-cyber no worries, sorry to hear that, take care brother. Hope you get well soon.

**Comment by jvsinghk** [comment]
@setheal would you like to contribute to this issue?

**Comment by maelannb** [comment]
@jvsinghk sure, I'll do it when I have a bit of time :)

**Comment by maelannb** [comment]
@devil-cyber take care mate

**Comment by jvsinghk** [comment]
@setheal Sure, let me know you want me to assign it to you.

**Comment by maelannb** [comment]
@jvsinghk You can assign it to me if it still opened

Btw, do you need to add `react-router` to the project ? I can make the server side configuration at the same time if needed

**Comment by jvsinghk** [comment]
@setheal Sure, how are you planning to use `react-router` in SpaceX?

**Comment by maelannb** [comment]
@jvsinghk it's just to know if you want me to add the static router server side :) I already created my pr but don't hesitate to ask me in the future if you need help with the router :)

**Comment by jvsinghk** [comment]
@setheal I have used `react-router` in amazon-clone to add routes for different components. Here as it's the single page. I was wondering how you plan to make use of it here. Sure, can you help me understand this concept of static router server-side rendering? So, I would be able to add to my other application. I would like to learn about it.

**Comment by maelannb** [comment]
@jvsinghk sure ! It's just a limitation of `react-router`, you need to use a different router on the server (called `StaticRouter`). I'm usually putting the `StaticRouter` around my top component on the server and a `BrowserRouter` in the root file of the client.  This way you can handle the routes server side and client side :)

**Comment by jvsinghk** [comment]
@setheal great! Thanks for the simple straight explanation. Will use the `StaticRouter` and `BrowserRouter` in the coming and existing projects. If possible, please add a resource to which can help me learn it's implementation.

**Comment by maelannb** [comment]
@jvsinghk Sure ! You can check this 3 links:
* [https://www.digitalocean.com/community/tutorials/react-react-router-ssr](https://www.digitalocean.com/community/tutorials/react-react-router-ssr)
* [https://reactrouter.com/web/guides/server-rendering](https://reactrouter.com/web/guides/server-rendering)
* [https://medium.com/javascript-in-plain-english/server-side-rendering-in-react-redux-ab0af31c9c4b](https://medium.com/javascript-in-plain-english/server-side-rendering-in-react-redux-ab0af31c9c4b)

Don't hesitate to contact me if you need any help :)

**Comment by jvsinghk** [comment]
@setheal Awesome! Sure, Tysm for the resources :)
